src/Database/PostgreSQL/TransactionalStore.hs

```diff
@@ -21,74 +22,97 @@ module Database.PostgreSQL.TransactionalStore
 import           Control.Applicative
 #endif
 import           Control.Monad.Reader
+import           Control.Monad.Trans.Control
 import           Data.Int
-import qualified Database.PostgreSQL.Simple         as Postgres
-import qualified Database.PostgreSQL.Simple.Transaction as Postgres.Transaction
-import qualified Database.PostgreSQL.Simple.Types   as PGTypes
+import qualified Database.PostgreSQL.Simple             as Postgres
 import           Database.PostgreSQL.Simple.FromRow
 import           Database.PostgreSQL.Simple.ToRow
+import qualified Database.PostgreSQL.Simple.Transaction as Postgres.Transaction
+import qualified Database.PostgreSQL.Simple.Types       as PGTypes
 
-newtype PGTransaction a =
-    PGTransaction (ReaderT Postgres.Connection IO a)
+newtype PGTransactionT m a =
+    PGTransactionT (ReaderT Postgres.Connection m a)
     deriving ( Functor
              , Applicative
              , Monad
-             , MonadIO
+             , MonadTrans
              , MonadReader Postgres.Connection
+             , MonadIO
              )
 
-runPGTransaction' :: MonadIO m
+type PGTransaction = PGTransactionT IO
+
+runPGTransaction' :: MonadBaseControl IO m
```

Comment:
Happy to provide a convenience function along the lines of 
```
runPGTransactionIO :: MonadIO m => PGTransactionT IO a -> Postgres.Connection -> m a
runPGTransactionIO = (IiftIO .) . runPGTransaction
```
but I wouldn't want to modify the the other functions, because `m ~ IO` does not allow us to embed effects.